### PR TITLE
Refactor CheckBox and StackedCheckBox components

### PR DESCRIPTION
**Pull request by kulyk**
This PR refactors `CheckBox` and `StackedCheckBox` components. The main motivation is to fix a11y issues, but it also aligns components with our current code style.

* refactors `CheckBox` to use `<input type="checkbox" />` under the hood. Currently it's just a div with `role="checkbox"` and click handlers
* refactors `StackedCheckBox` to use only one `CheckBox` input, and to render a stacked background as a different component. Currently, it just renders two checkboxes, so if you tab through the page, you'd focus on both of them which is pretty unexpected
* extends both components with a `label` prop. Now you can pass a string or a custom React component to display a label on the right side (clicking a label would lead to a `checked` change as if you click the checkbox itself)
* refactors `CheckBox` not to use deprecated color names
* code style: migrated to styled-components, prop-types position, .etc

### To Verify

1. Open `/_internal/components/checkbox` and `/_internal/components/stackedcheckbox`
2. Ensure every demo makes sense

Note: `StackedCheckBox` with a label renders weirdly in the StyleGuide, but looks as expected in the app itself. You can check a `FieldsPicker` example

1. Ask a question > Custom Question > Sample Dataset > Any Table
2. Click a "Columns" button on the right side of the "Data" panel (the one with a data source)
3. A popup with a list of table columns opens, look on the "Select All" checkbox in the top

### CODE REVIEW

File: frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js:606]: ❤️ 🎉 🥳

File: frontend/src/metabase/internal/components/ScratchApp.jsx
Comment (nemanjaglumac) [inline frontend/src/metabase/internal/components/ScratchApp.jsx:107]: Should this be in a template tag for translation?

**Comment by nemanjaglumac** [comment]
This is looking and working fantastic!
Great job, @kulyk 🚀

I checked (pun intended) collections and everything's working as expected. Maybe wait for someone else to review FE code as well before merging.

Also, we might want to wait before we ship 0.40 before merging larger portions refactored code.

File: .eslintrc
Comment (nemanjaglumac) [inline .eslintrc:69]: This is now showing additional console errors on every single page:
image

**Comment by gusaiani** [comment]
Nice direction this is taking!

File: frontend/src/metabase/components/CheckBox.styled.js
Comment (alxnddr) [inline frontend/src/metabase/components/CheckBox.styled.js:45]: When a checkbox is checked it makes the focus state not visible. Maybe it worth having the same focus outline for all colors from the UX perspective as well. What do you think?

Comment (alxnddr) [reply frontend/src/metabase/components/CheckBox.styled.js:45]: So maybe we won't also need to handle focus and blur manually

**Review by alxnddr — APPROVED** [review]
Overall looks great! I left one suggestion about the focused state, please take a look

File: frontend/src/metabase/components/CheckBox.styled.js
Comment (kulyk) [reply frontend/src/metabase/components/CheckBox.styled.js]: @alxnddr good catch, thank you. Fixed that, so the outline is displayed for the focused state as well. Still, not sure we can easily avoid manual focus handling until we have a more fresh version of `styled-components` :(

**Demo**

**Pull request merged**
